lib/natalie/compiler/pass0c.rb

```diff
@@ -53,6 +163,97 @@ def canary_array_max(exp, values)
         return exp.new(:if, exp.new(:call, first, :>, second), canary_array_max(exp, [first, *rest]), canary_array_max(exp, [second, *rest]))
       end
 
+      # FIXME: These are pretty repetitive, maybe do these with meta programming
```

Comment:
How about `canary_const_op(exp, values, op)` where op is a symbol and you use send on the first operand?

It'd look something 

```ruby
def canary_const_op(exp, values)
    _, (_, arg1), op, (_, arg2) = exp
   arg1.send op arg2
end

```

Reply:
sure I will try that